hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSEditLogOp.java

```diff
@@ -252,7 +252,7 @@ public void setTransactionId(long txid) {
   }
   
   public boolean hasRpcIds() {
-    return rpcClientId != RpcConstants.DUMMY_CLIENT_ID
+    return !Arrays.equals(rpcClientId, RpcConstants.DUMMY_CLIENT_ID)
```

Comment:
Thanks @WeisonWei , It is interesting catches. There will be some other place has the same un-corrected compares, Would you like to correct it at the same time?

Reply:
@Hexiaoqiao  I found no more similar problems after searching the project

Reply:
org.apache.hadoop.hdfs.server.namenode.FSEditLogOp#getClientId